### DISCUSSION TITLE
Query parameters in collections

### DISCUSSION THREAD

**Discussion by absid89** [usebruno/bruno#2428] [Q&A] [answered]
Hi, is there any way I can add a query parameter for all queries in a collection? 

![Screenshot 2024-06-09 130145](https://github.com/usebruno/bruno/assets/26306868/d08cbb4b-ffa2-4944-a6b2-2264469c7f7b)

**Comment by CodeShakingSheep** [comment, marked as answer]
Hi @absid89 ,
I just found a solution to your question. In a collection you can write a pre request script like this.

```js
const queryParam = "page=3";
req.setUrl(req.getUrl().includes("?") ? req.getUrl() + "&" + queryParam : req.getUrl() + "?" + queryParam);
```

Collection pre request script:
![image](https://github.com/usebruno/bruno/assets/19874562/90eec1f1-eafb-4365-b6f6-b61e7352d8c5)

Request timeline with appended query parameter:
![image](https://github.com/usebruno/bruno/assets/19874562/0d682d5e-85f4-42b6-b414-479ee36419b0)

You can see that the query parameter defined in pre request script is appended to the end of the URL.

Hope this helps. Cheers.

**Reply by absid89** [reply]
Hi @CodeShakingSheep 
thanks for the solution, I also figured out how to do it almost the same way via script. I thought there was a more direct solution, strange that it is not available out of the box

**Reply by CodeShakingSheep** [reply]
I also expected this to be more out of the box. For my use case I wanted to have my global query parameters more organized. So, I added them as env vars prefixed with `GQP_`. Then in the pre script I put the global query parameter names in an array and appended them dynamically in a loop. This is my script.

```js
let queryParams = ""; 
const globalQueryParamNames = [ "GQP_PER_PAGE", "GQP_PAGE" ];
for (const el of globalQueryParamNames) {
  queryParams += (queryParams === "" ? "" : "&") + el.split("_").slice(1).join("_").toLowerCase() + "=" + bru.getEnvVar(el);
}
req.setUrl(req.getUrl().includes("?") ? req.getUrl() + "&" + queryParams : req.getUrl() + "?" + queryParam);
```
